doc/plutus-core-spec/builtins-future.tex

```diff
@@ -0,0 +1,413 @@
+    \endlastfoot
+%% G1
+    \TT{bls12\_381\_G1\_add}  &
+    $[ \ty{bls12\_381\_G1\_element}$,
+      \text{\; $\ty{bls12\_381\_G1\_element} ]$}
+      \text{\: $ \to \ty{bls12\_381\_G1\_element}$} & Group addition &  No & \\
+    \TT{bls12\_381\_G1\_neg}  &
+      $ [ \ty{bls12\_381\_G1\_element} ]$  \text{\;\; $\to \ty{bls12\_381\_G1\_element}$} & Group inverse  & No & \\
+    \TT{bls12\_381\_G1\_scalarMul}  &
+    $[ \ty{integer}$,
+      \text{\; $\ty{bls12\_381\_G1\_element} ]$}
+      \text{\: $ \to \ty{bls12\_381\_G1\_element}$} & $(n,a) \mapsto na$ &  No & \\
+    \TT{bls12\_381\_G1\_equal}  &
+    $[ \ty{bls12\_381\_G1\_element}$,
+      \text{\; $\ty{bls12\_381\_G1\_element} ]$}
+      \text{\: $ \to \ty{bool}$} & $=$ &  No & \\
+    \TT{bls12\_381\_G1\_hashToGroup}  &
+    $[ \ty{bytestring}, \ty{bytestring}]$
+      \text{\: $ \to \ty{bls12\_381\_G1\_element}$} & $\hash_{G_1}$ &  Yes & \ref{note:hashing-into-group}\\
+    \TT{bls12\_381\_G1\_compress}  &
+    $[\ty{bls12\_381\_G1\_element}]$
+      \text{\: $ \to \ty{bytestring}$} & $\compress_{G_1}$  &  No & \ref{note:group-compression}\\
+    \TT{bls12\_381\_G1\_uncompress}  &
+    $[ \ty{bytestring}]$
+      \text{\: $ \to \ty{bls12\_381\_G1\_element}$} & $\uncompress_{G_1}$  &  Yes & \ref{note:group-uncompression}\\
+    \hline 
+%% G2
+    \TT{bls12\_381\_G2\_add}  &
+    $[ \ty{bls12\_381\_G2\_element}$,
+      \text{\; $\ty{bls12\_381\_G2\_element} ]$}
+      \text{\: $ \to \ty{bls12\_381\_G2\_element}$} & $(a,b) \mapsto a+b$ &  No & \\
+    \TT{bls12\_381\_G2\_neg}  &
+      $ [ \ty{bls12\_381\_G2\_element} ]$  \text{\;\; $\to \ty{bls12\_381\_G2\_element}$} & $a \mapsto -a$  & No & \\
+    \TT{bls12\_381\_G2\_scalarMul}  &
+    $[ \ty{integer}$,
+      \text{\; $\ty{bls12\_381\_G2\_element} ]$}
+      \text{\: $ \to \ty{bls12\_381\_G2\_element}$} & $(n,a) \mapsto na$ &  No & \\
+    \TT{bls12\_381\_G2\_equal}  &
+    $[ \ty{bls12\_381\_G2\_element}$,
+      \text{\; $\ty{bls12\_381\_G2\_element} ]$}
+      \text{\: $ \to \ty{bool}$} & $=$ &  No & \\
+    \TT{bls12\_381\_G2\_hashToGroup}  &
+    $[ \ty{bytestring}, \ty{bytestring}]$
+      \text{\: $ \to \ty{bls12\_381\_G2\_element}$} & $\hash_{G_2}$  &  Yes & \ref{note:hashing-into-group}\\
+    \TT{bls12\_381\_G2\_compress}  &
+    $[\ty{bls12\_381\_G2\_element}]$
+      \text{\: $ \to \ty{bytestring}$} & $\compress_{G_2}$  &  No & \ref{note:group-compression}\\
+    \TT{bls12\_381\_G2\_uncompress}  &
+    $[ \ty{bytestring}]$
+      \text{\: $ \to \ty{bls12\_381\_G2\_element}$} & $\uncompress_{G_2}$  &  Yes & \ref{note:group-uncompression}\\
+    \hline 
+    \TT{bls12\_381\_millerLoop}  &
+    $[ \ty{bls12\_381\_G1\_element}$,
+      \text{\; $\ty{bls12\_381\_G2\_element} ]$}
+    \text{\: $ \to \ty{\MlResult}$} & $e$ &  No & \ref{note:pairing}\\
+    \TT{bls12\_381\_mulMlResult}  &
+    $[ \ty{\MlResult}, \ty{\MlResult}] \to \ty{\MlResult}$ & Group multiplication & No & \ref{note:pairing}\\
```

Comment:
Just nitpicking., but I wonder why not write the H-group multiplication in the same way as the multiplication for $G_1$ and $G_2$

Reply:
> Just nitpicking., but I wonder why not write the H-group multiplication in the same way as the multiplication for G₁ and G₂.

OK, I've changed that (although it's a different operation: in G₁ and G₂ it's repeated addition and in H it's genuine group multiplication).